File: algorithms/programmers/43165.js
function createCalculation(numbers) {
  if (numbers.length === 1) {
    return [numbers[0], -numbers[0]];
  }
  const rest = numbers.slice(1);
  const calculations = [
    ...createCalculation(rest).map((calculation) => calculation + numbers[0]),
    ...createCalculation(rest).map((calculation) => calculation - numbers[0]),
  ];
  return calculations;
}

function solution(numbers, target) {
  return createCalculation(numbers).filter(
    (calculation) => calculation === target
  ).length;
}

File: algorithms/programmers/43165.py
def solution(numbers, target):
    count = 0

    def recursive(index, total):
        nonlocal count

        # BC
        if index == len(numbers):
            if total == target:
                count += 1
            return

        # 재귀함수 호출
        recursive(index + 1, total + numbers[index])
        recursive(index + 1, total - numbers[index])

    recursive(0, 0)
    return count
